From 83efed4bf5361baa16f93945e8dee344d35c4152 Mon Sep 17 00:00:00 2001
From: srihars1 <sriharsha.govinda.raj@intel.com>
Date: Thu, 2 Apr 2015 13:34:20 +0530
Subject: [PATCH] [EM/BQ24292] Set inlimit on charger recovery from exception

This patch set inlimit after charger has
recovered from voltage exceptions.

Change-Id: Idc9abce23d25672b8517a78dfd7f0dc4fef5a483
imin-jira: https://jira01.devtools.intel.com/browse/IMINAN-18536
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-7678
Signed-off-by: Yashodhan Kurtadkar <yashodhan.kurtadkar@intel.com>
Signed-off-by: srihars1 <sriharsha.govinda.raj@intel.com>
---
 drivers/power/bq24192_charger.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index c25b47f..f854902 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -1694,8 +1694,20 @@ static void bq24192_irq_worker(struct work_struct *work)
 	 * disconnect, re-init completion so that setting INLIMIT would be
 	 * delayed till VBUS is detected.
 	 */
-	if (reg_status & SYSTEM_STAT_VBUS_HOST)
+	if (reg_status & SYSTEM_STAT_VBUS_HOST) {
 		complete(&chip->vbus_detect);
+
+		/* On recovering from voltage exception after VBUS
+		 * is detected and supply voltage recovered,
+		 * inlimit has to be set to proper value
+		 */
+		if (reg_status & SYSTEM_STAT_PWR_GOOD &&
+			reg_status & SYSTEM_STAT_FAST_CHRG) {
+				mutex_lock(&chip->event_lock);
+				bq24192_set_inlmt(chip, chip->inlmt);
+				mutex_unlock(&chip->event_lock);
+		}
+	}
 	/*
 	 * If chip version is BQ24297, then it can detect DCP charger
 	 * cable type also. Hence complete the VBUS detect wait if
-- 
1.9.1

